### ISSUE TITLE
resource not found azureud

### ISSUE THREAD

**Issue by Dean-Christian-Armada** [rexray/rexray#1245]
"Err": "starting container failed: error while mounting volume '/var/lib/docker/plugins/8ea1a61da26d6d27fe182733d1a2312ed0ac07cfbff3c6294b709c2dda0e0062/rootfs': VolumeDriver.Mount: docker-legacy: Mount: myvolume: failed: resource not found",

Is the error above when I'm trying to run a countainer with a `rexray/azureud` volume driver. I am attempting to run it in a Docker Swarm. The node that actually installed the volume is just working fine when trying to mount the volume but when attempted with other nodes. The error above rises.

**Comment by clintkitson** [comment]
Hello @Dean-Christian-Armada,

Are you trying to mount the volume simultaneously to multiple hosts? If so that would explain this error. REX-Ray can only mount block volumes to one host at a time.

**Comment by Dean-Christian-Armada** [comment]
@clintkitson , If that's the case then I think the statement in the photo below is a little bit deceiving.

<img width="849" alt="screen shot 2018-07-26 at 06 54 22" src="https://user-images.githubusercontent.com/9837087/43231893-dd99c1ec-90a0-11e8-954b-83814fdb18ee.png">

The way I understand it, it is a centralized static storage of your Swarm.

**Comment by holms** [comment]
@clintkitson I've set delay 5seconds between container update/restart and parallelism set to 1, and it still fails to start container with message: ```
VolumeDriver.Mount: docker-legacy: Mount: tools-npm-register: failed: resource not found
```

**Comment by clintkitson** [comment]
I don't believe this is accurate since block volumes cannot be mounted to multiple hosts unless the filesystem on top of the devices supports clustering (distributed locking). EXT4/XFS does not do this so generally the use case would not work. However, you can expect that a volume is mounted/unmounted in an orchestrated fashion when it is requested on different hosts. This is what REX-Ray does when Docker makes the proper volume driver mount/unmount requests. The expectation is that Swarm is notifying the underlying docker engine's at the right times to move volumes where necessary.

**Comment by holms** [comment]
I couldn't force swarm to have 3 replicas on multiple hosts to use same rexray volume. I throws errors. Moving to flocker as it based on Ceph also has EBS backend, although to set it up is a nightmare, but there's consistency in there.
